### PR TITLE
Add heuristics tests for gfx90a architecture

### PR DESCRIPTION
**Pull request by marekjg**

### CODE REVIEW

**Review by cderb — APPROVED** [review]
LGTM

**Review by junliume — APPROVED** [review]

**Comment by atamazov** [comment]
@junliume @marekjg This PR has formatting problems.

NB: I had to fix these in my `ci_perf_test_performance` after merging fresh develop into it, so merge conflicts may happen in the future.

File: test/gtest/tuna_net.cpp [hunk 1]
```diff
@@ -101,26 +131,27 @@ void TestSolverPredictionModel(miopen::conv::ProblemDescription& problem,
 #else
     std::ignore = problem;
     std::ignore = expected_solver;
+    std::ignore = device_architecture
```
Comment (atamazov) [inline test/gtest/tuna_net.cpp:134]: 🐛 Missing semicolon. I suspect this wasn't built with MIOPEN_ENABLE_AI_IMMED_MODE_FALLBACK=Off

**Comment by atamazov** [comment]
> @junliume @marekjg This PR has formatting problems.
> 
> NB: I had to fix these in my `ci_perf_test_performance` after merging fresh develop into it, so merge conflicts may happen in the future.

Fixed in <a href=https://github.com/ROCm/MIOpen/pull/2785/commits/0257f0fc17e9867f026899e682fac6789ade30b0>PR 2785, commit 0257f0fc1</a>

File: test/gtest/tuna_net.cpp (hunk 1 above)
Comment (atamazov) [reply test/gtest/tuna_net.cpp:134]: Fixed in <a href=https://github.com/ROCm/MIOpen/pull/2785/commits/6bee000efac77beff0bd2d55c7c88af0c7a9cccc>PR 2785, commit 6bee000ef</a>

**Comment by junliume** [comment]
@atamazov Thanks! Somehow I thought this PR has passed all CI stages but apparently I have made a mistake. We fixed in develop branch and hope it should not cause any more problems there.

**Pull request merged**